src/Model/User.hs

```diff
@@ -188,9 +190,36 @@ updateUserPreview UserUpdate{..} user = user
 --------------------------------------------------------------------------------
 -- Database functions
 
+-- | Fetch all users, ordered by descending user id.
+fetchAllUsersDB :: DB [Entity User]
+fetchAllUsersDB =
+      select $
+      from $ \user -> do
+      orderBy [desc (user ^. UserId)]
+      return user
+
 fetchUsersInDB :: [UserId] -> DB [Entity User]
 fetchUsersInDB user_ids = selectList [UserId <-. user_ids] []
 
+-- | Fetch all users with their "project infos" (mapping from project name and
+-- handle to the set of user roles).
+fetchAllUserProjectInfosDB :: DB (Map UserId (Map (Text, Text) (Set Role)))
+fetchAllUserProjectInfosDB = fmap go query
+  where
+    go :: [(Value UserId, Value Text, Value Text, Value Role)] -> Map UserId (Map (Text, Text) (Set Role))
+    go = M.fromListWith (M.unionWith S.union) . map f
```

Comment:
I'm not sure what's better about it. The surface differences are probably fused away anyway.